docs/kms-apis/kms-apis.md

```diff
@@ -0,0 +1,118 @@
+# KMS APIs for AWS Nitro Enclaves
+
+To enable AWS Nitro Enclaves to exchange secrets with AWS KMS using
+cryptographic attestation, AWS KMS added a Recipient field to the following API calls:
+ * [Decrypt](./Decrypt.md)
+ * [GenerateDataKey](./GenerateDataKey.md)
+ * [GenerateRandom](./GenerateRandom.md)
+
+The Recipient field accepts a [RecipientRequest structure](./RecipientRequest.md)
+containing an `AttestationDocument` that includes a public key. After using the
+CMK to decrypt the `CiphertextBlob` in the request, or generating the Data Key or
+Random number, AWS KMS re-encrypts the resulting `Plaintext` blob using the public
+key and the specified encryption algorithm. Then it returns the data encrypted under
+the public key in the `CiphertextForRecipient` field in the response. This data
+can be decrypted only by using a private key that never leaves the enclave.
+In this case, the `Plaintext` field is eliminated.
+Examples for the new [request structure](#example-requests) and
+[response structure](#example-response) are provided below.
+
+`CiphertextForRecipient` is a RecipientInfo structure, as described in
+[RFC5652 Section 6](https://tools.ietf.org/html/rfc5652#section-6) encoded as Base64,
+and needs to be decrypted with the private key corresponding to the public key
+added in the AttestationDocument.
+
+These APIs allow an AWS Nitro Enclave to decrypt, encrypt (via `GenerateDataKey`) data
+and generate a random number if they match the AWS KMS policy set in place, and provide a 
+response from AWS KMS that only the enclave can use.
+
+Cryptographic attestation allows restricting access to an AWS KMS key based on a set of
+additional conditions that can be verified only by providing a valid `AttestationDocument`.
+Access is then given if the `AttestationDocument` is validly signed by the Nitro hypervisor, 
+it contains a public key and the PCRs listed in the AttestationDocument match the PCRs
+in the policy. An [example policy](#example-policy) is provided below, which allows the role `ROLE_ARN` to
+use the decrypt operation if it can provide a valid AttestationDocument containing
```

Comment:
For the sake of being consistent I would put every occurrence of "AttestationDocument" between back quotes